### PR TITLE
[Matrix][SYCL] Add support for bf16's wi_element

### PR DESCRIPTION
**Pull request by yubingex007-a11y**
Previously, we didn't take bf16's operation into consideration. Now we add support for bf16's wi_element and when we have operations of bf16, we extend to fp32 and do the operations.

### CODE REVIEW

**Comment by alexbatashev** [comment]
@AlexeySotkin @dkhaldi I suggest you creating a team on github and adding it as a code owner for this code to https://github.com/intel/llvm/blob/sycl/.github/CODEOWNERS, much like ESIMD extension is added.

@yubingex007-a11y I'll approve this PR if Alexey or Dounia approve

File: sycl/include/sycl/ext/oneapi/matrix/matrix-jit.hpp
Comment (AlexeySotkin) [inline sycl/include/sycl/ext/oneapi/matrix/matrix-jit.hpp:455]: I think we should specialize this class on `bfloat16` type.

File: sycl/include/sycl/ext/oneapi/matrix/matrix-jit.hpp
Comment (AlexeySotkin) [inline sycl/include/sycl/ext/oneapi/matrix/matrix-jit.hpp:517]: Why we can't use `__spirv_ConvertBF16ToFINTEL` and `__spirv_ConvertFToBF16INTEL` here?

File: sycl/include/sycl/ext/oneapi/matrix/matrix-jit.hpp
Comment (dkhaldi) [reply sycl/include/sycl/ext/oneapi/matrix/matrix-jit.hpp:455]: Alexey, we are adding the support of bfloat16 type as the next step in a separate PR.
This PR is to extend the support of the matrix bf16 type represented by unsigned short.
so we are not changing the way we treat bf16 right now.
Instead, we plan on keeping both: unsigned short as bf16, then SYCL bfloat16 type.
We will keep both until the SYCL bfloat16 makes it to the specification (it is experimental right now)

Comment (yubingex007-a11y) [reply sycl/include/sycl/ext/oneapi/matrix/matrix-jit.hpp:455]: @AlexeySotkin @dkhaldi For now we still consider uint16_t as bf16 in matrix and in the future we will support sycl's bf16.

File: sycl/include/sycl/ext/oneapi/matrix/matrix-jit.hpp
Comment (dkhaldi) [inline sycl/include/sycl/ext/oneapi/matrix/matrix-jit.hpp:454]: add a comment that uint16_t here is representing bf16 type as we have been doing for the previous matrix functions.
Since the AMX and DPAS implementations don't support uint16_t, this should raise no problem. 
Add a comment that the plan is to move towards SYCL bfloat16 as it makes it to the specification.

File: sycl/include/sycl/ext/oneapi/matrix/matrix-jit.hpp
Comment (yubingex007-a11y) [reply sycl/include/sycl/ext/oneapi/matrix/matrix-jit.hpp:517]: @AlexeySotkin because SPV_INTEL_bfloat16_conversion is not added to OpenCL CPU RT yet. so i write a workaround

File: sycl/include/sycl/ext/oneapi/matrix/matrix-jit.hpp
Comment (dkhaldi) [inline sycl/include/sycl/ext/oneapi/matrix/matrix-jit.hpp:454]: Some rewording:
// Note that similarly to the other matrix functions, uint16_t is used here to represent bf16 type. Since the AMX and DPAS implementations don't support uint16_t, this interpretation is possible. This design choice was made before the introduction of SYCL experimental bfloat16 type. Our plan is to move towards using the SYCL bfloat16. But since it is still experimental, we will probably keep both uint16 interpretation and SYCL bfloat16.

File: sycl/include/sycl/ext/oneapi/matrix/matrix-jit.hpp
Comment (dkhaldi) [inline sycl/include/sycl/ext/oneapi/matrix/matrix-jit.hpp:512]: Some rewording:
// We use here the following functions for conversion (bf16=>fp32 and fp32=>bf16). This is a workaround until we are able to use __spirv_ConvertFToBF16INTEL and __spirv_ConvertBF16ToFINTEL once these are supported in the CPU backend

**Review by dkhaldi — APPROVED** [review]
LGTM

**Comment by yubingex007-a11y** [comment]
ping? @intel/llvm-reviewers-runtime

**Comment by yubingex007-a11y** [comment]
@alexbatashev Hi, could you approve and merge it?

**Review by alexbatashev — APPROVED** [review]
@bader or @vladimirlaz or @dm-vodopyanov can you please merge?

File: sycl/include/sycl/ext/oneapi/matrix/matrix-jit.hpp
Comment (Nuullll) [reply sycl/include/sycl/ext/oneapi/matrix/matrix-jit.hpp:527]: FYI. I'm working on supporting SYCL_INTEL_bf16_conversion capability on CPU: https://github.com/intel-innersource/applications.compilers.source.ocl-rt/pull/301

**Pull request merged**
